Makefile

```diff
@@ -0,0 +1,126 @@
+SHELL := /bin/bash
+.SHELLFLAGS := -eu -o pipefail -c ${SHELLFLAGS}
+MVN_OPTS="-Dmaven.test.skip=true"
+OS:=$(shell . /etc/lsb-release &>/dev/null && echo $${DISTRIB_ID} ||  uname  )
+db_type:=postgres
+DBURL := $(shell case "${db_type}" in\
+			 (postgres) printf "postgres://postgres:postgres@localhost/autoscaler?sslmode=disable"; ;; \
+ 			 (mysql) printf "root@tcp(localhost)/autoscaler?tls=false"; ;; esac)
+
+$(shell mkdir -p target)
+
+.PHONY: check-type
+check-db_type:
+	@case "${db_type}" in\
+	 (mysql|postgres) echo "Using bd:${db_type}"; ;;\
+	 (*) echo "ERROR: db_type needs to be one of mysql|postgres"; exit 1;;\
+	 esac
+
+.PHONY: init-db
+init-db: check-db_type start-db build-db target/init-db-${db_type}
+target/init-db-${db_type}:
+	@./scripts/initialise_db.sh ${db_type}
+	@touch $@
+
+.PHONY: githooks
+githooks: target/githooks
+target/githooks: target/precommit-${OS}
+	@precommit install
+	@touch $@
+target/precommit-Darwin:
+	@which pre-commit &> /dev/null || brew install pre-commit
+	@touch $@
+target/precommit-Ubuntu:
+	@echo " - not installing $@"
+target/precommit-Linux:
+	@echo " - not installing $@"
+
+.PHONY: install-cli
+install-cli: target/install-cli-${OS}
+target/install-cli-Darwin:
+	@bosh --version &> /dev/null || brew install cloudfoundry/tap/bosh-cli
+	@cf --version &> /dev/null || brew install cloudfoundry/tap/cf-cli@7
+	@bbl --version &> /dev/null || brew install bbl
+	@[ $(shell cf plugins | grep AutoScaler | wc -l) -gt 1  ] || cf install-plugin -f -r CF-Community app-autoscaler-plugin
+	@touch $@
+target/install-cli-Ubuntu:
+	@echo " - not installing $@"
+target/install-cli-Linux:
+	@echo " - not installing $@"
+
+.PHONY: init
+init: target/init
+target/init:
+	@make -C src/autoscaler buildtools
+	@touch $@
+
+.PHONY: clean
+clean:
+	@rm target/* &> /dev/null || echo "# Already clean"
+
+.PHONY: build-db
+build-db: target/build-db
```

Comment:
can we change the .github/workflows/mysql|postgres.yml to call these new targets to ensure they are tested

Reply:
done :)